src/sst/core/baseComponent.cc

```diff
@@ -535,188 +536,121 @@ uint8_t BaseComponent::getComponentInfoStatisticEnableLevel(const std::string& s
-                    statParams = si.params;
-                    // Get the load level from the component
-                    // stat_load_level = curr_info->component->getComponentInfoStatisticEnableLevel(si.name);
-                    nameFound = true;
-
-                    all_enabled = (std::string(STATALLFLAG) == si.name);
-                    enable_level = curr_info->getStatisticLoadLevel();
-                    break;
-                }
-            }
-        }
-        next_info = curr_info->parent_info;
-    } while ( curr_info->canInsertStatistics() && !nameFound );
-
-    // hack for now to make sure that an explicitly enabled stat is
-    // always loaded regardless of load level.
-    if ( !all_enabled ) enable_level = 0xfe;
-
-    // Did we find a matching enable name?
-    if (false == nameFound) {
-        statGood = false;
-        out.verbose(CALL_INFO, 1, 0, " Warning: Statistic %s is not enabled in python script, statistic will not be enabled...\n", fullStatName.c_str());
-    }
-
-    if (true == statGood) {
-        // Check that the Collection Rate is a valid unit type that we can use
-        if ((true == collectionRate.hasUnits("s")) ||
-            (true == collectionRate.hasUnits("hz")) ) {
-            // Rate is Periodic Based
-            statCollectionMode = StatisticBase::STAT_MODE_PERIODIC;
-        } else if (true == collectionRate.hasUnits("event")) {
-            // Rate is Count Based
-            statCollectionMode = StatisticBase::STAT_MODE_COUNT;
-        } else if (0 == collectionRate.getValue()) {
-            // Collection rate is zero and has no units
-            // so we just dump at beginning and end
-            collectionRate = UnitAlgebra("0ns");
-            statCollectionMode = StatisticBase::STAT_MODE_PERIODIC;
-        } else {
-            // collectionRate is a unit type we dont recognize
-            out.fatal(CALL_INFO, 1, "ERROR: Statistic %s - Collection Rate = %s not valid; exiting...\n", fullStatName.c_str(), collectionRate.toString().c_str());
-        }
+void
+BaseComponent::configureCollectionMode(Statistics::StatisticBase* statistic, const SST::Params& params, const std::string& name)
+{
+  StatisticBase::StatMode_t statCollectionMode = StatisticBase::STAT_MODE_COUNT;
+  Output& out = getSimulation()->getSimulationOutput();
+  std::string statRateParam = params.find<std::string>("rate", "0ns");
+  UnitAlgebra collectionRate(statRateParam);
+
+  //make sure we have a valid collection rate
+  // Check that the Collection Rate is a valid unit type that we can use
+  if (collectionRate.hasUnits("s") || collectionRate.hasUnits("hz")) {
+      // Rate is Periodic Based
+      statCollectionMode = StatisticBase::STAT_MODE_PERIODIC;
+  } else if (collectionRate.hasUnits("event")) {
+      // Rate is Count Based
+      statCollectionMode = StatisticBase::STAT_MODE_COUNT;
+  } else if (collectionRate.getValue() == 0) {
+      // Collection rate is zero and has no units
```

Comment:
This isn't technically true.  The getValue() check is only looking for a value of zero, regardless of the units.  All we know is that it doesn't have one of the three units we are looking for.  Probably not a big deal since it's only a comment, but the comment is wrong.

Reply:
Yeah, we can definitely fix the comment. I think that comment might pre-date the changes, actually.